lib/src/main/java/graphql/nadel/engine/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -561,11 +560,25 @@ private class Factory(
         }
     }
 
-    private fun <T : GraphQLType> getUnderlyingType(overallType: T): T? {
+    /**
+     * Gets the underlying type for an [GraphQLObjectType]
+     *
+     * The [childField] is there in case the [overallType] is an operation type.
+     * In that case we still need to know which service's operation type to return.
+     */
+    private fun getUnderlyingType(
+        overallType: GraphQLObjectType,
+        childField: GraphQLFieldDefinition,
+    ): GraphQLObjectType? {
         val renameInstruction = makeTypeRenameInstruction(overallType as? GraphQLDirectiveContainer ?: return null)
-        val service = definitionNamesToService[overallType.name]
-            ?: error("Unknown service for type: ${overallType.name}")
         val underlyingName = renameInstruction?.underlyingName ?: overallType.name
+
+        val fieldCoordinates = makeFieldCoordinates(overallType, childField)
+
+        val service = definitionNamesToService[overallType.name]
+            ?: coordinatesToService[fieldCoordinates]
+            ?: error("Unable to determine service for $fieldCoordinates")
```

Comment:
Main fix is here.
Before we would do type name -> service.
This doesn't work for operation types.
So we now fall back to field name -> service.